src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -211,6 +213,26 @@ private void trimMap(NavigableMap<Long, ?> map, int maxSize) throws Exception {
         }
     }
 
+    /**
+     * Deletes the MetricsDB entries in the map till the size of the map is equal to maxSize. The actual on-disk
+     * files is deleted ony if the config is not set or set to true.
+     */
+    private void trimDatabases(NavigableMap<Long, MetricsDB> map, int maxSize) throws Exception {
+        boolean deleteDBFiles = PluginSettings.instance().shouldCleanupMetricsDBFiles();
+        // Remove the oldest entries from the map, upto maxSize.
+        while (map.size() > maxSize) {
+            Map.Entry<Long, MetricsDB> lowestEntry = map.firstEntry();
+            if (lowestEntry != null) {
+                MetricsDB value = lowestEntry.getValue();
+                value.remove();
+                if (deleteDBFiles) {
+                    value.deleteOnDiskFile();
+                }
+                map.remove(lowestEntry.getKey());
```

Comment:
NIT: For safety, it would be better to remove the key from the map and then delete the value/file. This is just in case there is a race with QueryMetricsHandler.